qiskit/providers/models/backendproperties.py

```diff
@@ -105,25 +114,191 @@ def __init__(self, qubits, gate, parameters, **kwargs):
+    All properties which are provided are provided optionally. These properties may describe
+    qubits, gates, or other general properties of the backend.
     """
 
-    def __init__(self, backend_name, backend_version, last_update_date,
-                 qubits, gates, general, **kwargs):
+    def __init__(self,
+                 backend_name: str,
+                 backend_version: str,
+                 last_update_date: datetime.datetime,
+                 qubits: List[List[Nduv]],
+                 gates: List[Gate],
+                 general: List[Nduv],
+                 **kwargs):  # pylint: disable=missing-param-doc
+        """Initialize a BackendProperties instance.
+
+        Args:
+            backend_name: Backend name.
+            backend_version: Backend version in the form X.Y.Z.
+            last_update_date: Last date/time that a property was updated.
+            qubits: System qubit parameters.
+            gates: System gate parameters.
+            general: General parameters.
+        """
         self.backend_name = backend_name
         self.backend_version = backend_version
         self.last_update_date = last_update_date
+        self.general = general
         self.qubits = qubits
         self.gates = gates
-        self.general = general
+
+        self._qubits = {}
+        for qubit, props in enumerate(qubits):
+            formatted_props = {}
+            for prop in props:
+                value = self._apply_prefix(prop.value, prop.unit)
+                formatted_props[prop.name] = (value, prop.date)
+                self._qubits[qubit] = formatted_props
+
+        self._gates = {}
+        for gate in gates:
+            if gate.gate not in self._gates:
+                self._gates[gate.gate] = {}
+            formatted_props = {}
+            for param in gate.parameters:
+                value = self._apply_prefix(param.value, param.unit)
+                formatted_props[param.name] = (value, param.date)
+            self._gates[gate.gate][_to_tuple(gate.qubits)] = formatted_props
 
         super().__init__(**kwargs)
+
+    def gate_property(self,
+                      gate: str,
+                      qubits: Union[int, Iterable[int]] = None,
+                      name: str = None) -> Tuple[Any, datetime.datetime]:
+        """
+        Return the property of the given gate.
+
+        Args:
+            gate: Name of the gate.
+            qubits: The qubit to find the property for.
+            name: Optionally used to specify which gate property to return.
+
+        Returns:
+            Gate property as a tuple of the value and the time it was measured.
+
+        Raises:
+            PulseError: If the property is not found or name is specified but qubit is not.
+        """
+        try:
+            result = self._gates[gate]
+            if qubits is not None:
+                result = result[_to_tuple(qubits)]
```

Comment:
Why `_to_tuple`? We are trying to move away from there.

Reply:
Move away from where, exactly? Can you rephrase this please?

Reply:
writing a qubit as a tuple is something we are trying to avoid doing. We prefer `Qubit` over`(QuantumRegister, integer)`

Reply:
I don't want a dependency from providers -> Qubit. We aren't writing a qubit as a tuple of register, integer, we are writing them as a tuple of all the qubit indices (their labels). This is only so that the dictionary keys are always the same, whether the user provides `[0]` or `0` or `[0, 1]` or `(0, 1)` to get info from the backend. Similar for `PulseDefaults`, we don't want Pulse to be dependent on Circuit's `Qubit`s, we just need a consistent way to hash them with minimal overhead in the code or by the user in order to get cmd_def Schedules out